### PR TITLE
Fixes tool softlock in creating a proper anvil

### PR DESCRIPTION
**Pull request by Malorn-Deslor**
#### Summary
Bugfixes "Fixes tool softlock in creating a proper anvil"

#### Purpose of change
With recent changes in #68386, we entered a strange situation where all steel/iron anvils could only be made via railroad tracks using angle grinders. This is, of course, a valid way to make an anvil, but humanity had anvils long before railroads or angle grinders. There were also other changes which resulted in a softlock in tools, where a hammer 3 was required to make the tools which would be needed to make a hammer with hammer 3.

#### Describe the solution
I added in a second, more time-consuming method to creating a crude anvil, and made creating a proper anvil possible with grinding qualities and drilling qualities, rather than specifically requiring power tools. I also tweaked the chisel recipe to required anvil 2 not anvil 3, since it is required in order to make anvil 3. Finally, I added a 'crude hammer' which is a rough forged steel hammer with hammer 3 quality (using anvil 2 not anvil 3). But it has none of the prying abilities since it is not a clawed hammer. This removes the problem of needing a hammer to make a hammer. 

#### Describe alternatives you've considered
Obviously this softlock can be bypassed using bronze anvils, which are currently very easy to make, assuming you can find the tin. However this is not ideal, since realistically many steel faced anvils could be created before angle grinders ever were thought up. I'm also not entirely certain bronze anvils should be quite as high quality as proper steel ones, but I'm not an expert and there may be metallurgical tricks to making bronze with a harder facing or something like that.

#### Testing
Tested the entire crafting progression on a fresh save. Works fine and given enough skill and salvaged steel, you can once again enter the metal age.

#### Additional context
Someone with a better knowledge of toolcrafting should work through the various recipes that need anvil 3. Since anvil 2 is basically just a big chunk of hammer hardened steel it should work well for anything that doesn't use the shaping advantages provided by the horn, etc.

### CODE REVIEW

**Comment by Fris0uman** [comment]
Your justifications for this PR don't hold up. We want real things to behave close to reality, "I can make ANVIL 2 because I need it for game progression" is not a good justification. You need to show that irl the recipe you suggested would produce the level of functionality you gave them.

**Comment by Malorn-Deslor** [comment]
> Your justifications for this PR don't hold up. We want real things to behave close to reality, "I can make ANVIL 2 because I need it for game progression" is not a good justification. You need to show that irl the recipe you suggested would produce the level of functionality you gave them.

Sure, my irl justification is fairly simple: humanity made anvils, they did not require anvils to already exist to make anvils. People made hammers which were good enough to make good anvils without anvils of the quality which required very good hammers to make.

Requiring an angle grinder, or making a situation where the tools required to make a high quality anvil require that same high quality anvil is fundamentally impossible from a realism perspective. Creating a situation where the only way to create a high quality anvil is from a railroad track is obviously absurd.

Respectfully, I think you misunderstood my point about progression, it's not for 'gameplay 'progression that we need these recipes, it's because such things were possible and thus our game, being realistic, should provide a similar route. Realistically, it should be even easier, since any large chunk of steel would be far superior to the semi-steel anvils which were used throughout most of human history.

:::

Now let's talk about how actual anvils were made. The actual creation process for anvils in pre-industrial times involved layering red-hot iron and hammering repeatedly to build a block of hardened wrought iron (essentially welding it using heat and force). This is layered with carbon dust on the upper face and hot hammered to create a semi-steel facing. Finally chisels and grinding methods are used to shape the anvil horn, using a drill to notch a couple of places to make blacksmithing tricks easier.

There are plenty of documents which detail this method, and even a few youtube videos showing the method in action. The process is physically challenging but not complex. It does require a lot of heavy hammering and a decent forge to get the metal hot enough. Honestly, I'm not at all sure it even requires a casting method at all, though the progress could be accelerated using sand casting on individual parts.

**Comment by Drew4484** [comment]
> > Your justifications for this PR don't hold up. We want real things to behave close to reality, "I can make ANVIL 2 because I need it for game progression" is not a good justification. You need to show that irl the recipe you suggested would produce the level of functionality you gave them.
> 
> Sure, my irl justification is fairly simple: humanity made anvils, they did not require anvils to already exist to make anvils. People made hammers which were good enough to make good anvils without anvils of the quality which required very good hammers to make.
> 
> Requiring an angle grinder, or making a situation where the tools required to make a high quality anvil require that same high quality anvil is fundamentally impossible from a realism perspective. Creating a situation where the only way to create a high quality anvil is from a railroad track is obviously absurd.
> 
> Respectfully, I think you misunderstood my point about progression, it's not for 'gameplay 'progression that we need these recipes, it's because such things were possible and thus our game, being realistic, should provide a similar route. Realistically, it should be even easier, since any large chunk of steel would be far superior to the semi-steel anvils which were used throughout most of human history.
> 
> :::
> 
> Now let's talk about how actual anvils were made. The actual creation process for anvils in pre-industrial times involved layering red-hot iron and hammering repeatedly to build a block of hardened wrought iron (essentially welding it using heat and force). This is layered with carbon dust on the upper face and hot hammered to create a semi-steel facing. Finally chisels and grinding methods are used to shape the anvil horn, using a drill to notch a couple of places to make blacksmithing tricks easier.
> 
> There are plenty of documents which detail this method, and even a few youtube videos showing the method in action. The process is physically challenging but not complex. It does require a lot of heavy hammering and a decent forge to get the metal hot enough. Honestly, I'm not at all sure it even requires a casting method at all, though the progress could be accelerated using sand casting on individual parts.

Before industrialization, anvils were made like any other metal tool: worse and with a lot more labor. Stone anvils were replaced by bronze anvils, which were replaced by iron, and then steel. What we don't include in the game is the extreme labor requirements for forging an iron or steel anvil without tools. A dozen skilled individuals gathered around a massive piece of iron that vaguely resembles an anvil, hammering it as hard and as fast as possible to shape it before it cooled, often swapping out to a new worker as they became exhausted and overheated. This can be avoided by casting an anvil, but a cast anvil is worse by every measure other than ease of crafting. 

If you want to bypass the Free Steel Everywhere and don't have the patience to find an already crafted hammer or anvil, add more low-tech tools. Add an alternate recipe for a crude anvil that replaces power tools with hours of labor. Add cast anvils. Add more hammer recipes. Just don't redo the work that already went into updating anvils for the sake of playing in the woods.

**Comment by Malorn-Deslor** [comment]
> Before industrialization, anvils were made like any other metal tool: worse and with a lot more labor. Stone anvils were replaced by bronze anvils, which were replaced by iron, and then steel. What we don't include in the game is the extreme labor requirements for forging an iron or steel anvil without tools. A dozen skilled individuals gathered around a massive piece of iron that vaguely resembles an anvil, hammering it as hard and as fast as possible to shape it before it cooled, often swapping out to a new worker as they became exhausted and overheated. This can be avoided by casting an anvil, but a cast anvil is worse by every measure other than ease of crafting.
> 
> If you want to bypass the Free Steel Everywhere and don't have the patience to find an already crafted hammer or anvil, add more low-tech tools. Add an alternate recipe for a crude anvil that replaces power tools with hours of labor. Add cast anvils. Add more hammer recipes. Just don't redo the work that already went into updating anvils for the sake of playing in the woods.

Uhh...that's exactly what I did, created a second crude anvil recipe that has more time required. And I am not talking about being in the woods, I am talking about using all that 'free steel' to make an anvil, which would be much easier than using iron. I also added a hammer recipe...just as you describe. Why are you suggesting I do something which I already did originally? Did you think I just reverted the change or something?

Also...you need to check your data, and I'd really like to see that source. Making an anvil was not a job for dozens of men, nor did it take weeks. It was a job that generally was done by a blacksmith master and his journeyman (making the anvil for said journeyman in most cases), and the process was labor intensive, not time intensive. But hey, you want more time? I'm fine with that, what time do you think it should take? Throw me some numbers. 

Perhaps you are thinking about the massive anvils that were created, which likely did require dozens of men and weeks of work? Because sure, there are massive historical anvils weighing thousands of pounds. But those are the minority and certainly are not what is being crafted in CDDA.

https://www.youtube.com/watch?v=Uc3jcUm0m2Y

This is literally the first search result for making an anvil: two guys, forge welding. The result is on the smaller side of anvils, certainly, but I can point you to hundreds of similar anvil across centuries. They even manage to make a horn without any grinding tool, which is frankly impressive.

::::

Don't discard centuries of human crafting because you like hunting around for an anvil or not crafting your own tools. If we are going to be realistic, it's the knowledge that would incredibly rare. People dedicated years to learning blacksmithing, with teachers. But CDDA allows you to gain skills insanely quickly, and that was not my design decision, maybe it's the blob connecting you to a collective unconscious or something weird like that. But sure, I have no idea how the PC learns the crap they learn so easily. But that's not relevant to what crafting recipes should exist.

**Comment by KHeket** [comment]
And don't forget about cast-molding method of making anvils like in medieval - raw iron ore was smelted in a chimney shaped furnace into a spongy bloom and then wrought (beaten) into a workable shape and alloy. The first iron anvils were simply very large blooms worked into a block with at least one flat surface.
In this case you can use metal objects and have benefit. Ore can have impurities, what will make metal worse, but when you will use modern already cleared from impurities metal objects - your anvil will be much more better
Its quite similar to the simple making crude anvil. And you don't need any very durable object to make simple metal objects
Also, here is the video of making 100kg anvil vie cast-molding
https://www.youtube.com/watch?v=THh8xgqUS44&ab_channel=AusMetalCasting
So for 26 kg anvil you need x4 less sized items
And don't forget about Viking anvils, which was much simpler than classic anvil, but that was enough for lots of metal works
image

**Comment by GuardianDll** [comment]
just to be sure, the current recipe uses 10 h of work, has `activity_level` 4 out of 5, doesn't require any forging, require hammering 2 (which is equal to stone hammer) 

in your video you show two people, who forge anvil, using forge, proper steel tools and another anvil

(also your recipe will overwrite the recipe with angle_grinder and railroad track because lack of `"id_suffix"`)

**Comment by Malorn-Deslor** [comment]
> And don't forget about cast-molding method of making anvils like in medieval - raw iron ore was smelted in a chimney shaped furnace into a spongy bloom and then wrought (beaten) into a workable shape and alloy. The first iron anvils were simply very large blooms worked into a block with at least one flat surface. In this case you can use metal objects and have benefit.

A huge number of items have anvil 1 for this exact reason, I think. As to cast anvils, I couldn't really justify it with research I could find. In theory, CDDA's recipe is using steel, which as you say should result in a better anvil even if cast, but at the same time there are a lot of people who have problems with modern cast anvils shattering, etc. I had thought (wrongly it seemed) to stick to something that was well understood and non-controversial.

**Comment by Fris0uman** [comment]
Here's some context and previous discussions https://github.com/CleverRaven/Cataclysm-DDA/pull/68386  , the PR itself links to more stuff, and suggests what can be added on top of it.

**Comment by Malorn-Deslor** [comment]
> just to be sure, the current recipe uses 10 h of work, has `activity_level` 4 out of 5, doesn't require any forging, require hammering 2 (which is equal to stone hammer)
> 
> in your video you show two people, who forge anvil, using forge, proper steel tools and another anvil
> 
> (also your recipe will overwrite the recipe with angle_grinder and railroad track because lack of `"id_suffix"`)

As I said, first video. And yes, obviously modern people are not going to use stone anvils, nor more crude tools. But realistically, is anyone really saying that making something with less than ideal tools is impossible? In reality you might break your stone tools a couple times, but you'd have an anvil in the end. And makeshift hammers are just chunks of steel on a stick, more than enough to hammer other things. Keep in mind the angle grinding recipe uses hammer 2 as well, and angle grinders do not get around the requirement to face-harden.

But I take your point, it should require anvil 1, and I am open to increasing the time involved, perhaps a high estimate of 20 hours? You may also notice I boosted the difficulty to 6 for that same reason. As to activity, I took that from all the other metalworking, and I think 4 out of 5 is reasonable since this is a task that can be done for hours.

As to the forge...thank you, I was copying from the other one and I didn't realize it didn't include the use of a forge, since it ought to. It should obviously require a forge and I'll get that sorted out immediately. Thank you again for noticing it was lacking. Can't believe all the prior anvil recipes didn't require any heating of the metal...

**Comment by Malorn-Deslor** [comment]
> I don’t really have a horse in this race, so I think this pr is probably fine. But one of the suggestions from the previous work on anvils was that someone should audit the ANVIL requirements in all the recipes. Does making a chisel really require ANVIL 3? Chisels don’t need holes punched into them, so you wouldn’t need a pritchel hole. And they don’t have curves or bends, so you don’t need the horn either. Maybe that recipe should simply be lowered to ANVIL 1 or ANVIL 2 instead.

I agree with that audit completely, but I lack the personal background to do it. I've done basic blacksmithing, but anything more than hardening, shaping, and attaching is past my skill level by a wide margin. I honestly don't know what it would take to make things like a metal file set, for example.

I did lower chisel specifically due to what you mentioned above, however. It is something that merely requires a lot of hardening.

**Comment by Drew4484** [comment]
> > Before industrialization, anvils were made like any other metal tool: worse and with a lot more labor. Stone anvils were replaced by bronze anvils, which were replaced by iron, and then steel. What we don't include in the game is the extreme labor requirements for forging an iron or steel anvil without tools. A dozen skilled individuals gathered around a massive piece of iron that vaguely resembles an anvil, hammering it as hard and as fast as possible to shape it before it cooled, often swapping out to a new worker as they became exhausted and overheated. This can be avoided by casting an anvil, but a cast anvil is worse by every measure other than ease of crafting.
> > If you want to bypass the Free Steel Everywhere and don't have the patience to find an already crafted hammer or anvil, add more low-tech tools. Add an alternate recipe for a crude anvil that replaces power tools with hours of labor. Add cast anvils. Add more hammer recipes. Just don't redo the work that already went into updating anvils for the sake of playing in the woods.
> 
> Uhh...that's exactly what I did, created a second crude anvil recipe that has more time required. And I am not talking about being in the woods, I am talking about using all that 'free steel' to make an anvil, which would be much easier than using iron. I also added a hammer recipe...just as you describe. Why are you suggesting I do something which I already did originally? Did you think I just reverted the change or something?
> 
> Also...you need to check your data, and I'd really like to see that source. Making an anvil was not a job for dozens of men, nor did it take weeks. It was a job that generally was done by a blacksmith master and his journeyman (making the anvil for said journeyman in most cases), and the process was labor intensive, not time intensive. But hey, you want more time? I'm fine with that, what time do you think it should take? Throw me some numbers.
> 
> Perhaps you are thinking about the massive anvils that were created, which likely did require dozens of men and weeks of work? Because sure, there are massive historical anvils weighing thousands of pounds. But those are the minority and certainly are not what is being crafted in CDDA.
> 
> https://www.youtube.com/watch?v=Uc3jcUm0m2Y
> 
> This is literally the first search result for making an anvil: two guys, forge welding. The result is on the smaller side of anvils, certainly, but I can point you to hundreds of similar anvil across centuries. They even manage to make a horn without any grinding tool, which is frankly impressive.
> 
> ::::
> 
> Don't discard centuries of human crafting because you like hunting around for an anvil or not crafting your own tools. If we are going to be realistic, it's the knowledge that would incredibly rare. People dedicated years to learning blacksmithing, with teachers. But CDDA allows you to gain skills insanely quickly, and that was not my design decision, maybe it's the blob connecting you to a collective unconscious or something weird like that. But sure, I have no idea how the PC learns the crap they learn so easily. But that's not relevant to what crafting recipes should exist.

TLDR: Large quality anvils are very hard to make if you don't have one already.
As mentioned, see https://github.com/CleverRaven/Cataclysm-DDA/pull/64595#issuecomment-1506525069 

The main point I'm circling here is the difference between a large heavy flat object, an anvil-shaped piece of metal, and an actual anvil you would use for heavy metalworking tasks. The video you reference shows two people working (in a modern, high-temp forge with modern high-quality tools) to forge  weld pieces of metal into a small anvil. The end result is a perfectly serviceable small anvil, with some caveats.

Dimensions: The anvil they create is relatively small. This impacts both working surface (you need a larger surface for larger crafts, look for videos of swords or armor being crafted for an idea of the scale) and the weight, which is needed for stability. Weight /should/ be around 50:1 of anvil/hammer, and a larger craft will require a heavier hammer.

Construction: The anvil shown is forge-welded, which isn't a bad thing necessarily, but is inferior to a single stock piece of metal (like a railroad track!). This impacts durability but also a factor called 'rebound'. Rebound is a measure of how well the anvil reflects energy back to the craft. Soft metals have terrible rebound, cast metals have not quite as bad rebound, additive/buildup is better, and a single monolithic piece of metal is ideal. In game we do not simulate this as we disallow high quality anvils of poor material. Poor rebound acts as a cap on your craft hardness. If you're hammering a thin sheet of copper, you don't need much force and a flat piece of wood will do the trick. If you're hammering 2mm of steel into a breastplate, you need to hit it very hard, and if the anvil is lacking in rebound you are wasting energy with every strike, the steel is not being shaped correctly, and perhaps worst of all to a cataclysm survivor: this costs time and fuel to reheat and correct flaws.
 
Ideally you have a single piece of good steel heated all the way through and then hammered into shape. This is significantly more difficult if you don't have access to or cannot make a single piece of metal in the size you want your anvil to be, so you make compromises. If you don't have access to a trip hammer, you have to rely on manual labor to perform the same task, which is where the team of people with hammers exhausting themselves comes in for making a heavy anvil. I should clarify that this is talking about very large anvils that are not needed for most crafts, even in game, which is why we have an anvil 4 quality.

**Comment by db48x** [comment]
> I did lower chisel specifically due to what you mentioned above, however. It is something that merely requires a lot of hardening.

Yes, I saw. However, have you considered that lowering the anvil quality for the chisel recipe might be the only thing required to let people craft a proper anvil? If you can get a chisel then you don’t need to craft a crude anvil.

> TLDR: Large quality anvils are very hard to make if you don't have one already. As mentioned, see [#64595 (comment)](https://github.com/CleverRaven/Cataclysm-DDA/pull/64595#issuecomment-1506525069)
> 
> The main point I'm circling here is the difference between a large heavy flat object, an anvil-shaped piece of metal, and an actual anvil you would use for heavy metalworking tasks. The video you reference shows […] a perfectly serviceable small anvil, with some caveats.
>
> […]

Yea, everyone here already knows all of that crap. Sounds to me like you are agreeing that people can just make an anvil any time they want, but it’ll be ANVIL 2 rather than ANVIL 3 for the reasons you listed. Which is exactly what @Malorn-Deslor has added in this very PR.

File: data/json/recipes/recipe_others.json [hunk 1]
```diff
@@ -3054,18 +3059,34 @@
...
+    "activity_level": "ACTIVE_EXERCISE",
+    "result": "anvil_crude",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "difficulty": 6,
+    "time": "20 h",
+    "autolearn": true,
+    "qualities": [ { "id": "HAMMER", "level": 2 }, { "id": "GRIND", "level": 2 }, { "id": "ANVIL", "level": 1 } ],
+    "tools": [ [ [ "crucible", -1 ], [ "crucible_clay", -1 ], [ "casting_mold", -1 ] ], [ [ "forge", 200 ], [ "oxy_torch", 200 ] ] ],
```
Comment (kevingranade) [inline data/json/recipes/recipe_others.json:3073]: If this is supposed to represent forge welding a bunch of chunks together then the crucible is a distraction. If you're imagining a process where you start with iron chunks and make regular ingots out of them with casting and forge weld them together, then make the input be ingots and we can have a separate recipe for that.
Also, if we're representing forge welding 27 kg of iron together these requirements are not representative of the properties of a typical forge. It's not going to have either the volume or heat capacity for the task you're describing here.
Finally, forge welding requires forging. There's an anvil 1 requirement, but that is intended for a small but irregular forging task. You need something to act as a "large crude anvil".

File: data/json/recipes/recipe_others.json [hunk 2]
```diff
@@ -3054,18 +3059,34 @@
...
+    "proficiencies": [ { "proficiency": "prof_metalworking" }, { "proficiency": "prof_blacksmithing" } ],
+    "components": [ [ [ "steel_lump", 28 ] ] ]
+  },
   {
     "type": "recipe",
     "activity_level": "MODERATE_EXERCISE",
     "result": "anvil_crude",
+    "id_suffix": "angle_grind",
```
Comment (kevingranade) [inline data/json/recipes/recipe_others.json:3081]: Differentiating based on the rail stock is better than a tool it uses.

File: data/json/recipes/recipe_others.json [hunk 3]
```diff
@@ -3054,18 +3059,34 @@
...
     "category": "CC_OTHER",
     "subcategory": "CSC_OTHER_TOOLS",
     "skill_used": "fabrication",
     "difficulty": 3,
     "time": "4 h",
     "autolearn": true,
-    "qualities": [ { "id": "HAMMER", "level": 2 } ],
-    "tools": [ [ [ "angle_grinder", 200 ] ] ],
+    "qualities": [ { "id": "HAMMER", "level": 2 }, { "id": "ANVIL", "level": 1 } ],
+    "tools": [ [ [ "angle_grinder", 200 ] ], [ [ "forge", 100 ], [ "oxy_torch", 100 ] ] ],
```
Comment (kevingranade) [inline data/json/recipes/recipe_others.json:3089]: An oxy torch is fine as an alternate for cutting the rail,  but what's the forge for? Hardening? One of the reasons to use rails is they're already sufficiently hard.

File: data/json/items/tool/workshop.json [hunk 4]
```diff
@@ -452,6 +452,21 @@
     "weapon_category": [ "BATONS" ],
     "melee_damage": { "bash": 9 }
   },
+  {
+    "id": "hammer_crude",
+    "type": "TOOL",
+    "name": { "str": "crude hammer" },
+    "description": "A simple crude steel hammer.  It is not as sophisticated or useful as a properly made alternative but it does get the job of hammering in nails well enough.",
```
Comment (kevingranade) [inline data/json/items/tool/workshop.json:459]: The description is very illustrative.  Hammering in nails,  fine.  Forging,  no.

**Review by kevingranade — CHANGES_REQUESTED** [review]

**Comment by kevingranade** [comment]
In broad strokes, the problem with this PR is you're not connecting the dots between "this should be possible" and "how exactly is it done".

A massive forge weld is a lot more of a change than just upping the fuel and time requirements. As for boot strapping, a large forge weld is the single operation. We have in the game that most needs a high quality anvil as a base and you're just skipping over that.

File: data/json/recipes/tools/tools_hand.json [hunk 5]
```diff
@@ -263,6 +264,30 @@
     "tools": [ [ [ "drift", -1 ] ] ],
     "components": [ [ [ "2x4", 1 ], [ "stick", 2 ] ] ]
   },
+  {
+    "type": "recipe",
+    "activity_level": "BRISK_EXERCISE",
+    "result": "hammer_crude",
```
[the remark this answers is not in the document]
Comment (Malorn-Deslor) [reply data/json/recipes/tools/tools_hand.json:270]: As far as I know it is not being represented as being cast, it requires an anvil, just not anvil 3. You need a hammer, just not hammer 3. The normal hammer is a modern claw-style hammer, which would require a lot of shaping and notching, which would require an anvil 3. The crude hammer is more of a traditional blacksmith hammer, being a large hardened broad hammer, no claw, no prying, etc. Making such a hammer is far easier than making one that curves.

https://encrypted-tbn1.gstatic.com/shopping?q=tbn:ANd9GcSfeOwZASNr_DSSfKAfluHdaFjYD0eFNAh_92OY89ur6jlPTdeg9yLhJFLu2pA1M53Qyp-8h2iKfRUYT5sfq2FGDqiO6qrQDHb9i_4yVUQS-Fa-H7lgjgtLauBbdRfOKhqsGm1-kqYQAM8&usqp=CAc

https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRUFO6sLGI_fjUOAty2CajYeENx4lbqyUfxvA&usqp=CAU

So basically, it's a simpler hammer, but probably more suited for blacksmithing than the claw hammer would be. You basically take the makeshift hammer, which is a lump of steel on a stick, and you heat it up in a forge and hammer it a lot to harden it. The key is the forge and a flat surface. Why did you think it was cast, just because of the crucible? Because the crucible is used for nearly all smithing currently, despite not being realistically necessary for a lot of forging work.

File: data/json/recipes/tools/tools_hand.json [hunk 6]
```diff
@@ -160,8 +161,8 @@
       [ "textbook_weparabic", 5 ]
     ],
     "using": [ [ "forging_standard", 3 ], [ "steel_tiny", 3 ] ],
-    "qualities": [ { "id": "ANVIL", "level": 3 }, { "id": "HAMMER", "level": 3 }, { "id": "GRIND", "level": 2 } ],
-    "//": "Specifically avoids blacksmithing requirement groups so that it's craftable without the metal fileset - this process would still be possible, just realistically take slightly longer.",
+    "qualities": [ { "id": "ANVIL", "level": 2 }, { "id": "HAMMER", "level": 3 }, { "id": "GRIND", "level": 2 } ],
+    "//": "Specifically avoids blacksmithing requirement groups so that it's craftable without the metal fileset - this process would still be possible, just realistically take slightly longer. Also avoids anvil 3 requirement since it is a direct requirement of making an proper anvil",
```
[the remark this answers is not in the document]
Comment (Malorn-Deslor) [reply data/json/recipes/tools/tools_hand.json:165]: I didn't write the first half of that note, someone else did, I just added the anvil part. I can clarify, however. Making a chisel is just hammer down a bar of steel into a flat tip, then grinding the flat tip until it's sharp. It would need need anything but a flat durable surface such as that provided by anvil 2, no curving, no notching.

File: data/json/recipes/recipe_others.json (hunk 1 above)
Comment (Malorn-Deslor) [reply data/json/recipes/recipe_others.json:3073]: I agree the crucible is a distraction, but every single metalworking recipe wants to use the crucible. It would it easier to get the steel into convenient sizes, but hardly necessary. I'm happy to drop it, just assumed it was policy to keep it for anything involving metal.

As to the forge, keep in mind that you are not heating the entire anvil in the forge. You heat each part and add it, so each chunk of steel is heated then added to the existing part. The 'welding' is hammer welding, where you hammer metal while red-hot and it welds together.

https://web.archive.org/web/20160303175903/http://www.abana.org/downloads/controlled_hand_forging/CHF_10.pdf

The kicker to this method is that you can work with small parts of iron and 'build' the anvil without needing to reheat the entire thing at any point. It's ideal for exactly the situation you describe, one with a limited forge and limited anvil. The heat capacity...I mean the forge can melt steel for casting, if it can do that, it can heat steel red-hot relatively easily.

If you'd like, we can add in a specific requirement for the 'anvil 1' such as a large rock or steel frame? Would that work? I agree you'd want as large a flat surface as possible.

File: data/json/recipes/recipe_others.json (hunk 2 above)
Comment (Malorn-Deslor) [reply data/json/recipes/recipe_others.json:3081]: Sure.

File: data/json/recipes/recipe_others.json (hunk 3 above)
Comment (Malorn-Deslor) [reply data/json/recipes/recipe_others.json:3089]: To soften enough to bend, was my thinking. Most videos I've seen using rail stock cut the stock, heat it with a blowtorch or forge, and then hammer it into a broader shape. Same reason I added a bit of anvil in there. The Hammer was already included, so I assumed they were following that method.

Trying to shape hardened steel with a hammer without heating it would risk fracturing it.

File: data/json/items/tool/workshop.json (hunk 4 above)
Comment (Malorn-Deslor) [reply data/json/items/tool/workshop.json:459]: I'm find changing the description. The thing is, creating a smithing hammer is more about metal quality than anything else. They actually aren't hard to make, they tend to be blocky and require only a little shaping on either end to create a flat and rounded surface respectively. Making a claw hammer would be far more difficult.

**Comment by Malorn-Deslor** [comment]
> In broad strokes, the problem with this PR is you're not connecting the dots between "this should be possible" and "how exactly is it done".
> 
> A massive forge weld is a lot more of a change than just upping the fuel and time requirements. As for boot strapping, a large forge weld is the single operation. We have in the game that most needs a high quality anvil as a base and you're just skipping over that.

I think the part where we are parting company is this seems very obvious to me, and clearly it isn't to other people. So I'll try to backtrack. Let's talk about why an anvil is useful, and for what. Modern anvils have a horn, hardy hole, pritchel hole, and a heel. Those are wonderful for making more complex shapes, notching, adding holes, etc. I would not like to make any sort of a curved tool without a horn.

But let's step back to what we are trying to do. Our crude anvil has no curves, it is a flat block of face hardened steel. It is literally just a flat rock, but more durable. Literally it would be possible to just take a large block of iron, dust it with flux, take a blow-torch to it, and face harden it. Wouldn't be fancy, but you'd have a great working surface. Or you can build one, per the video.

Another step back, you have your flat big rock, hopefully made of granite or something similar. This also works very well as an anvil, except you are wearing down the stone constantly and it will crack if you hit it wrong. But you can smith anything you can do with a block of steel with a block of stone. We don't model tool durability, which is the huge gap between the two. You also have to cut or find the stone every time you round out the 'anvil' which gets tiresome. (There is also an effort gap because rock doesn't provide rebound the way good iron or steel does, but that is about how tiring it is to smith, rather than what is possible.)

This is the core problem. This whole 'anvil 1' 'anvil 2' thing is a game progression thing, it's not realistic in the first place. If it were then all those huge rocks could be made into 'anvil 2' just but chipping them into having a flat surface. But I get it, we don't model durability or ease of use, so letting you use a huge rock as an anvil just as good as a proper steel one would be unrealistic. At the same time, soft-locking your ability to make such an anvil isn't realistic either.

A lovely example of someone trying out a stone anvil forging process below:

https://www.youtube.com/watch?v=VyVVn3F2DSk

As you can see, it's possible to do quite a lot with just a block of stone, and not even that large of one. For an anvil...yeah, you need a larger block of stone, not arguing that at all. But even our 'large rocks' are only around 34 lbs. So...I dunno, we need bigger rocks? All the larger rocks are immobile, though.

**Comment by db48x** [comment]
> This is the core problem. This whole 'anvil 1' 'anvil 2' thing is a game progression thing, it's not realistic in the first place. If it were then all those huge rocks could be made into 'anvil 2' just but chipping them into having a flat surface. But I get it, we don't model durability or ease of use, so letting you use a huge rock as an anvil just as good as a proper steel one would be unrealistic. At the same time, soft-locking your ability to make such an anvil isn't realistic either.

It would be a little more work, but I think you could almost implement that if you really want to. Make a recipe for creating a hammer and chisels on an ANVIL 1 (which can be a random big rock in a field). Make a recipe using the hammer and chisel (and perhaps other tools such as grinders) and the big rock to turn the rock into an ANVIL 2 big rock in a field, located in the same spot. Then make a recipe for creating an ANVIL 3 anvil using the ANVIL 2 big rock in a field that turns it back into an ANVIL 1 big rock in a field as a byproduct. This step represents the rock face that you created wearing out.

The difficulty is that recipes only work on items. Items and furniture can be inter–converted, but only as a separate step. That second recipe would require as input not the ANVIL 2 rock furniture, but the ANVIL 2 rock item. It would work, but it would feel weird that you can pick up this anvil and use it as an item in one recipe, while all the other anvil recipes require the anvil to be placed as furniture.

Fixing that would require modifying the crafting process to allow furniture to be in the inputs and outputs of the recipe, as well as items. Of course furniture has a position as well as type, so make sure the JSON for the recipe specifies which furniture is transformed into which other furniture. It’s extra work, but it’s a lot less extra work than introducing durability and ease of use as overarching concepts that apply to all crafting interactions.

File: data/json/items/tool/workshop.json (hunk 4 above)
Comment (Malorn-Deslor) [reply data/json/items/tool/workshop.json:459]: Fixed the description and increased the weight a bit to represent a more conventional smithing hammer.

File: data/json/recipes/recipe_others.json (hunk 1 above)
Comment (Malorn-Deslor) [reply data/json/recipes/recipe_others.json:3073]: I boosted the time even higher, just to represent the work involved.

Sdded the ability to use a chunks or lumps, respectively. Removed crucible since there is not really a need for it as you pointed out. Also increased forge fuel use since it is a lot of steel.

File: data/json/recipes/recipe_others.json (hunk 3 above)
Comment (Malorn-Deslor) [reply data/json/recipes/recipe_others.json:3089]: On more research on RR anvils, it also looks like most people strongly recommend adding material to the rail support using hammer-welding. Though you can also apparently partly encase the anvil in sand or dirt, so it probably would work ok with a little work. Also the rail anvil 'works' even without these methods, it just is harder to use.

File: data/json/recipes/tools/tools_hand.json [hunk 7]
```diff
@@ -137,7 +137,8 @@
     "autolearn": true,
     "proficiencies": [ { "proficiency": "prof_metalworking" }, { "proficiency": "prof_toolsmithing" } ],
     "using": [ [ "forging_standard", 3 ] ],
-    "qualities": [ { "id": "ANVIL", "level": 3 }, { "id": "HAMMER", "level": 3 }, { "id": "GRIND", "level": 2 } ],
+    "qualities": [ { "id": "ANVIL", "level": 2 }, { "id": "HAMMER", "level": 3 }, { "id": "GRIND", "level": 2 } ],
+    "//": "Intentionally avoids using anvil 3, since this is required to make the chisel required to create a proper anvil",
```
[the remark this answers is not in the document]
Comment (Malorn-Deslor) [reply data/json/recipes/tools/tools_hand.json:141]: Updated.

File: data/json/recipes/tools/tools_hand.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (Malorn-Deslor) [reply data/json/recipes/tools/tools_hand.json:165]: Updated the note with details.
